File: portfolio-client/src/app/components/model/workItem.ts
import {Skill} from "./skill";
import {OtherPosition} from "./otherPosition";

export class WorkItem {
  title: string;
  employer: string;
  formerly: string;
  start: string;
  end: string;
  description: string;
  skillsUsed: Skill[];
  otherPositions: OtherPosition[];
  accomplishments: string;


  constructor(title: string, employer: string, formerly: string, start: string, end: string, description: string
    , skillsUsed: Skill[], otherPositions: OtherPosition[], accomplishments: string) {
    this.title = title;
    this.employer = employer;
    this.formerly = formerly;
    this.start = start;
    this.end = end;
    this.description = description;
    this.skillsUsed = skillsUsed;
    this.otherPositions = otherPositions;
    this.accomplishments = accomplishments;
  }

  static from(workItem: any) {
    const item = new WorkItem(workItem.title
      , workItem.employer
      , workItem.formerly
      , workItem.start
      , workItem.end
      , workItem.description
      , workItem.skillsUsed
      , workItem.otherPositions
      , workItem.accomplishments);
    return item;
  }
}

File: portfolio-server/src/main/java/com/maksudsharif/portfolioserver/model/WorkItem.java
package com.maksudsharif.portfolioserver.model;

import java.util.ArrayList;
import java.util.List;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class WorkItem {
  private String title;
  private String employer;
  private String formerly;
  private String start;
  private String end;
  private String description;
  @Builder.Default
  private List<Skill> skillsUsed = new ArrayList<>();
  private String accomplishments;
  @Builder.Default
  private List<OtherPosition> otherPositions = new ArrayList<>();
}
